mkosi/config.py

```diff
@@ -950,8 +966,13 @@ def _load_json(cls, s: Union[str, dict[str, Any], SupportsRead[str], SupportsRea
         else:
             raise ValueError(f"{cls.__name__} can only be constructed from JSON from strings, dictionaries and files.")
 
-        transformer = json_type_transformer(cls)
-        return {k: transformer(k, v) for k, v in j.items()}
+        value_transformer = json_type_transformer(cls)
+        def key_transformer(k: str) -> str:
+            if (s := SETTINGS_LOOKUP_BY_NAME.get(k)) is not None:
+                return s.dest
+            return "_".join(part.lower() for part in FALLBACK_NAME_TO_DEST_SPLITTER.split(k))
```

Comment:
Ditto, do we need the fallback for `MkosiConfig`?

Reply:
Same, `preset` is the culprit.